evap/evaluation/management/commands/anonymize.py

```diff
@@ -197,23 +201,75 @@ def anonymize_evaluations(self):
                     evaluation.name_de = name[0]
                     evaluation.name_en = name[1]
                 else:
-                    evaluation.name_de = "Evaluierung #" + str(i + 1)
-                    evaluation.name_en = "Evaluation #" + str(i + 1)
+                    evaluation.name_de = f"Evaluierung #{i + 1}"
+                    evaluation.name_en = f"Evaluation #{i + 1}"
                 evaluation.save()
 
-    def anonymize_questionnaires(self, lorem_ipsum):
-        # questionnaires = Questionnaire.objects.all()
-
+    def anonymize_questionnaires(self):
         self.stdout.write("REMINDER: You still need to randomize the questionnaire names...")
         self.stdout.write("REMINDER: You still need to randomize the questionnaire questions...")
 
+    def anonymize_answers(self, lorem_ipsum):
         self.stdout.write("Replacing text answers with fake ones...")
         for text_answer in TextAnswer.objects.all():
             text_answer.answer = self.lorem(text_answer.answer, lorem_ipsum)
             if text_answer.original_answer:
                 text_answer.original_answer = self.lorem(text_answer.original_answer, lorem_ipsum)
             text_answer.save()
 
+        self.stdout.write("Shuffling rating answer counter counts...")
+
+        contributions = Contribution.objects.all().prefetch_related("ratinganswercounter_set__question")
+        try:
+            self.stdout.ending = ""
+            progress_bar = ProgressBar(self.stdout, contributions.count())
+            for contribution_counter, contribution in enumerate(contributions):
+                progress_bar.update(contribution_counter + 1)
+
+                counters_per_question = defaultdict(list)
+                for counter in contribution.ratinganswercounter_set.all():
+                    counters_per_question[counter.question].append(counter)
+
+                for question, counters in counters_per_question.items():
+                    original_sum = sum(counter.count for counter in counters)
+
+                    missing_values = set(CHOICES[question.type].values).difference(set(c.answer for c in counters))
+                    missing_values.discard(NO_ANSWER)  # don't add NO_ANSWER counter if it didn't exist before
+                    for value in missing_values:
+                        counters.append(RatingAnswerCounter(question=question, contribution=contribution, answer=value, count=0))
+
+                    generated_counts = [random.random() for c in counters]
+                    generated_sum = sum(generated_counts)
+                    generated_counts = [round(count / generated_sum * original_sum) for count in generated_counts]
```

Comment:
I think floow + adding up is less code and easier to understand, so will probably change that.